File: rb/34.rb
def factorial n
  n < 2 ? 1 : (2..n).reduce(:*)
end

def facsum n
  n.to_s.chars.map { |c| factorial c.to_i }.reduce(:+)
end

p (3..42000).select { |n| facsum(n) == n }.reduce(:+)

File: py/34.py
def factorial(n):
    return 1 if n < 2 else reduce(lambda a, b: a * b,
                             xrange(2, n + 1))

def facsum(n):
    return reduce(lambda a, b: a + b,
             map(factorial, map(int, list(str(n)))))

print reduce(lambda a, b: a + b,
        filter(lambda n: facsum(n) == n,
          xrange(3, 42000)))
